### PR TITLE
Handle positive and negative bar stacks

### PR DESCRIPTION
**Pull request by philippjfr**
Correctly handles mixed positive and negative stacked bars as requested in https://github.com/ioam/holoviews/issues/1709.

### CODE REVIEW

File: holoviews/plotting/bokeh/chart.py [hunk 1]
```diff
@@ -794,10 +801,8 @@ def get_data(self, element, ranges, empty):
 
         # Iterate over stacks and groups and accumulate data
         data = defaultdict(list)
-        baselines = defaultdict(float)
+        baselines = defaultdict(lambda: {True: 0, False: 0})
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/chart.py]: I find this a little weird and I don't think the code would be much more complicated if you use:

```
defaultdict(lambda: {'positive': 0, 'negative': 0})
```

or

```
defaultdict(lambda: {'+ve': 0, '-ve': 0})
```

**Comment by philippjfr** [comment]
Also fixes https://github.com/ioam/holoviews/issues/1939 now.

**Comment by philippjfr** [comment]
Here's the plot supplied in the linked issue, will add a unit test for that now:

bokeh_plot

**Comment by philippjfr** [comment]
@jlstevens Ready to merge. I think my next PR will be to organize tests into modules.

**Comment by jlstevens** [comment]
Looks good. Merging.

**Pull request merged**